test/unit/orchestration/tftp_test.rb

```diff
@@ -30,10 +30,9 @@ def test_host_should_not_have_tftp
 
   def test_generate_pxe_template_for_build
     if unattended?
-      h = FactoryGirl.create(:host, :build => true,
-                             :operatingsystem => operatingsystems(:redhat),
```

Comment:
If you're removing these to work around validation when creating a managed host, I think you can simply switch this FG call to .build - which both keeps it neater, and significantly reduces the number of records created.